ee/tabby-ui/lib/experiment-flags.ts

```diff
@@ -39,18 +40,29 @@ class ExperimentFlagFactory {
   }
 
   defineHook() {
```

Comment:
why need the change?

Reply:
The reason is same as my comment before
> Modify the hook not using useLocalStorage or the next.js will got error of rendering not aligned between server and client

Before when I import the hook, it throwed an next.js error (state not aligned between server and client rendering), it make sense because localStorage only avilable in the client. so I add `useEffect` to make sure the localStorage data fetching only happens in the client